plugins/DataStreamCereal/test_publisher.py

```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/python
+
+# todo: ensure PYTHONPATH is your cloned openpilot dir5
+from cereal.messaging import PubMaster, new_message
+import math
+import random
+from time import sleep
+
+pm = PubMaster(['deviceState'])
+
+rate = 20.
+time = 0.0
+last_temp = 30.
+
+while True:
+  sleep(1 / rate)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

For mathematical calculations, we recommend using the `decimal` module instead of the `float` datatype where any of the following is required - 
 1. exact decimal representation
 2. control over precision
 3. control over rounding to meet legal or regulatory requirements
 4. tracking of significant decimal places
 5. applications where the user expects the results to match calculations done by hand

[Learn more](https://docs.python.org/3/tutorial/stdlib2.html#decimal-floating-point-arithmetic)

Similar issue at line numbers 17 and 20.